src/components/node-list/node-list-groups.js

```diff
@@ -1,122 +1,84 @@
 import NodeListGroup from './node-list-group';
 import NodeListRow from './node-list-row';
-import {
-  toggleNodeClicked,
-  toggleNodeHovered,
-  toggleNodesDisabled
-} from '../../actions/nodes';
 
 const storedState = loadState();
 
 const NodeListGroups = ({
-  onToggleNodeClicked,
-  onToggleNodesDisabled,
-  onToggleNodeHovered,
-  nodes,
-  nodeActive,
-  nodeSelected,
-  types
+  items,
+  sections,
+  groups,
+  searchValue,
+  onToggleGroupChecked,
+  onItemClick,
+  onItemMouseEnter,
+  onItemMouseLeave,
+  onItemChange
 }) => {
-  // Deselect node on Escape key
-  const handleKeyDown = event => {
-    utils.handleKeyEvent(event.keyCode, {
-      escape: () => onToggleNodeClicked(null)
-    });
-  };
-  useEffect(() => {
-    window.addEventListener('keydown', handleKeyDown);
-    return () => window.removeEventListener('keydown', handleKeyDown);
-  });
-
   const [collapsed, setCollapsed] = useState(storedState.groupsCollapsed || {});
 
   // Collapse/expand node group
-  const onToggleCollapsed = typeID => {
+  const onToggleGroupCollapsed = typeID => {
     const groupsCollapsed = Object.assign({}, collapsed, {
       [typeID]: !collapsed[typeID]
     });
     setCollapsed(groupsCollapsed);
     saveState({ groupsCollapsed });
   };
 
-  // Toggle node selection depending on whether it's already selected
-  const handleNodeSelection = nodeID => {
-    if (nodeSelected[nodeID]) {
-      onToggleNodeClicked(null);
-    } else {
-      onToggleNodeClicked(nodeID);
-    }
-  };
-
-  const renderTypeGroup = type => {
-    if (!nodes[type.id]) {
-      return null;
-    }
-    return (
-      <NodeListGroup
-        key={type.id}
-        onToggleCollapsed={onToggleCollapsed}
-        type={type}
-        collapsed={collapsed[type.id]}>
-        <ul className="pipeline-nodelist pipeline-nodelist--nested">
-          {nodes[type.id].map(node => (
-            <li key={node.id}>
-              <NodeListRow
-                active={nodeActive[node.id]}
-                checked={!node.disabled_node}
-                disabled={node.disabled_tag || node.disabled_type}
-                id={node.id}
-                label={node.highlightedLabel}
-                name={node.name}
-                onClick={() => handleNodeSelection(node.id)}
-                onMouseEnter={() => onToggleNodeHovered(node.id)}
-                onMouseLeave={() => onToggleNodeHovered(null)}
-                onChange={e => {
-                  onToggleNodesDisabled([node.id], !e.target.checked);
-                }}
-                selected={nodeSelected[node.id]}
-                type={node.type}
-              />
-            </li>
-          ))}
-        </ul>
-      </NodeListGroup>
-    );
-  };
-
-  return (
-    <Flipper flipKey={collapsed}>
-      <ul className="pipeline-nodelist">{types.map(renderTypeGroup)}</ul>
-    </Flipper>
-  );
+  return sections.map(section => (
+    <nav className="pipeline-nodelist-section kedro" key={section.name}>
```

Comment:
Why are the sections `<nav>` elements? You've got multiple navs, which feels odd to me.
![image](https://user-images.githubusercontent.com/1155816/95448013-8e0f3500-095a-11eb-863a-dc4275df9a07.png)

I'd be more inclined to have something like this:
![image](https://user-images.githubusercontent.com/1155816/95448062-a41cf580-095a-11eb-8aba-29a768576acd.png)

Reply:
Happy to entertain discussion about what should or should not be a `<nav>`, but these definitely feel like `<section>`s to me.

Reply:
Do you mean they're not `<nav>` since they're not links? As far as I know multiple `<nav>` are normal in principle, though you could argue that the whole sidebar is just one `<nav>`. The `<section>` element is usually intended for blocks of content, like sections of an article or a book. I didn't choose it here as navigation isn't usually classed as content, so I wouldn't say this change is needed.

Reply:
It was more just that they didn't seem to me like separate nav panels, they're just different sections in the same nav panel. Overall not a big issue, I just felt like `<nav><section/><section/></nav>` made more sense than `<div><nav/><nav/></div>`. I believe you can use a section in a nav, but `<nav><nav/><nav/></nav>` might make more sense, idk?

Reply:
I think it works fine as it is personally, but if you like we could change the root element to `nav` and also the `NodeListGroup` too, then it would be `nav` all the way down 🐢